packages/doguinteraction/doguinteractor.go

```diff
@@ -95,106 +103,99 @@ func emptyDoguNameError() error {
+
+	watchOptions := metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%s", doguName),
+	}
+	watcher, err := ddi.doguClient.Watch(timeoutCtx, watchOptions)
+	if err != nil {
+		cancelTimeout()
+		return fmt.Errorf("error starting watch for dogu %s: %w", doguName, err)
+	}
+
+	for event := range watcher.ResultChan() {
+		switch event.Type {
+		case watch.Error:
+			watcher.Stop()
+			cancelTimeout()
+			return fmt.Errorf("error in watch while waiting for start/stop: %v", event.Object)
+		default:
+			isInDesiredState, cErr := ddi.checkIfDoguInDesiredStopState(timeoutCtx, doguName)
+			if cErr != nil {
+				watcher.Stop()
+				cancelTimeout()
+				return fmt.Errorf("error checking dogu-state while waiting for start/stop: %w", cErr)
 			}
 
-			if stopWait || rolledOut {
-				stopWaitChannels(timeoutTimer, ticker)
+			if isInDesiredState {
+				watcher.Stop()
+				cancelTimeout()
 				return nil
 			}
-		case <-timeoutTimer.C:
-			ticker.Stop()
-			return fmt.Errorf("failed to wait for deployment %s rollout: timeout reached", doguName)
 		}
 	}
-}
-
-func (ddi *defaultDoguInterActor) doWaitForDeploymentRollout(ctx context.Context, doguName string) (rolledOut bool, stopWait bool, err error) {
-	logrus.Info(fmt.Sprintf("check rollout status for deployment %s", doguName))
-	deployment, getErr := ddi.clientSet.AppsV1().Deployments(ddi.namespace).Get(ctx, doguName, metav1.GetOptions{})
-	if getErr != nil {
-		return false, true, fmt.Errorf("failed to get deployment %s: %w", doguName, getErr)
-	}
 
-	isInCrashLoop, err := ddi.isDoguContainerInCrashLoop(ctx, doguName)
-	if err != nil || isInCrashLoop {
-		return false, true, err
-	}
+	cancelTimeout()
+	watcher.Stop()
 
-	if deployment.Spec.Replicas != nil && deployment.Status.UpdatedReplicas < *deployment.Spec.Replicas {
-		logrus.Info(fmt.Sprintf("waiting for deployment %q rollout to finish: %d out of %d new replicas have been updated", deployment.Name, deployment.Status.UpdatedReplicas, *deployment.Spec.Replicas))
-		return false, false, nil
-	}
-	if deployment.Status.Replicas > deployment.Status.UpdatedReplicas {
-		logrus.Info(fmt.Sprintf("waiting for deployment %q rollout to finish: %d old replicas are pending termination", deployment.Name, deployment.Status.Replicas-deployment.Status.UpdatedReplicas))
-		return false, false, nil
-	}
-	if deployment.Status.AvailableReplicas < deployment.Status.UpdatedReplicas {
-		logrus.Info(fmt.Sprintf("waiting for deployment %q rollout to finish: %d of %d updated replicas are available", deployment.Name, deployment.Status.AvailableReplicas, deployment.Status.UpdatedReplicas))
-		return false, false, nil
-	}
-	logrus.Info(fmt.Sprintf("deployment %q successfully rolled out", deployment.Name))
-	return true, true, nil
+	return fmt.Errorf("watch for dogu %s stopped: timeout reached: %v", doguName, context.Cause(timeoutCtx))
```

Comment:
```suggestion
	return fmt.Errorf("watch for dogu %s stopped: %v", doguName, context.Cause(timeoutCtx))
```

context cause already says that timeout has been reached: `watch for dogu postgresql stopped: timeout reached: timout reached waiting for dogu postgresql. Timeout was 0s`